helix-rest/src/main/java/org/apache/helix/rest/server/resources/helix/ResourceAssignmentOptimizerAccessor.java

```diff
@@ -190,42 +184,44 @@ private InputFields readInput(String content)
   private ClusterState readClusterStateAndValidateInput(String clusterId, InputFields inputFields)
       throws InvalidParameterException {
 
+    // One instance can only exist in one of the list in InstanceChange.
+    // Validate the intersection is empty.
+    for (String instance : inputFields.activatedInstances) {
```

Comment:
To be more generic, consider we may have more lists soon, let's use a temporary hashset to check if an instance is mentioned more than once in all the lists.

Reply:
I had a temporary hashset to check duplication when swapInstance is in the scope. Since we only have 2 keyword, won't having a separate set be a bit overkill?

Reply:
I'm pretty sure we will have more lists soon. Keeping a temp hashset is not a big overhead. Let's just add it to avoid refactoring.
Or if you want to do shortcut now and refactor later, then we should use stream() to simplify code.

Reply:
TFTR.
Using steam or using a temp hashset will need some extra space. But I agree it won't cause big overhead.
Updated.